src/routes/Signup.jsx

```diff
@@ -1,43 +1,99 @@
 import HeadingA from "../components/HeadingA";
 import TextBox from "../components/TextBox";
+import PasswordBox from "../components/PasswordBox";
+import EmailBox from "../components/EmailBox";
 import AnimatedButton from "../components/AnimatedButton";
-import { useLocation } from "react-router-dom";
-import { useEffect, useState } from "react";
+import { useState } from "react";
+import { useNavigate } from "react-router-dom";
+import api from "../api";
 
-function useQuery() {
-  return new URLSearchParams(useLocation().search);
-}
+const SIGNUP_URL = "/auth/signup";
 
 export default function Signup() {
-  const query=useQuery();
-  const userName=query.get('username')
+  const navigate = useNavigate();
 
   const [credentials, setCredentials] = useState({
-    username:userName,
-    password:"",
-    confirmPassword:""
-  })
+    username: "",
+    email: "",
+    password: "",
+    confirmPassword: "",
+  });
+
+  const handleCredentials = (e) => {
+    setCredentials({ ...credentials, [e.target.name]: e.target.value });
+  };
 
-  const handleCredentials=(e)=>{
-    setCredentials({...credentials,[e.target.name]:e.target.value})
-  }
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+    if (credentials.password !== credentials.confirmPassword) {
+      alert("Password does not match");
+    } else {
+      try {
+        const response = await api.post(
+          SIGNUP_URL,
+          JSON.stringify({
+            name: credentials.username,
+            email: credentials.email,
+            password: credentials.password,
+          }),
+          {
+            headers: {
+              "Content-Type": "application/json",
+            },
+            withCredentials: true,
+          }
+        );
 
-  useEffect(()=>{
-    if(!userName){
-      alert("Invalid Username")
+        if (response.status === 200) {
+          alert("Signup successful");
+          navigate("/verifyEmail");
```

Comment:
As we already know the email of the user from his/her input, the email can be passed directly in the path like `/verifyEmail?email=${email}`